src/main/java/org/culturegraph/mf/morph/maps/RestMap.java

```diff
@@ -43,14 +47,37 @@ public void setUrl(final String url) {
 	public String get(final Object key) {
 		final Matcher matcher = VAR_PATTERN.matcher(url);
 		try {
-			final URL url = new URL(matcher.replaceAll(key.toString()));
-			final URLConnection con = url.openConnection();
-			// TODO correctly read from connection!
-			return (String) con.getContent();
+			String urlString = matcher.replaceAll(key.toString());
+			return readFromUrl(urlString);
+		} catch (IOException | URISyntaxException e) {
+			// There was no data result for the given URL
+			return null;
+		}
+	}
 
-		} catch (final IOException e) {
-			throw new MetafactureException(e);
+	private String readFromUrl(final String url) throws IOException, URISyntaxException {
+		InputStream inputStream = new URL(new URI(url.replace(" ", "%20")).toASCIIString()).openConnection()
```

Comment:
Why is the `url.replace(" ", "%20")` necessary here? The test data does not seem to contain a space, or am I misunderstanding something?

If in general characters that need encoding should be supported here, wouldn't a general approach like calling `URLEncoder.encode(String s, String enc)` with `key.toString()` when building the URL be safer?

And can we get rid of the `.toASCIIString()` to support keys like `organisations?q=Köln`, if that makes any sense?